encoding/plain/plain.go

```diff
@@ -37,23 +38,67 @@ func (e *Encoding) EncodeBoolean(dst []byte, src []byte) ([]byte, error) {
 }
 
 func (e *Encoding) EncodeInt32(dst []byte, src []int32) ([]byte, error) {
-	return append(dst[:0], unsafecast.Int32ToBytes(src)...), nil
+	if cpu.IsBigEndian {
+		srcLen := len(src)
+		byteEnc := make([]byte, (srcLen * 4))
+		idx := 0
+		for k := 0; k < srcLen; k++ {
+			binary.LittleEndian.PutUint32(byteEnc[idx:(4+idx)], uint32((src)[k]))
+			idx += 4
+		}
+		return append(dst[:0], (byteEnc)...), nil
+	} else {
+		return append(dst[:0], unsafecast.Int32ToBytes(src)...), nil
+	}
 }
 
 func (e *Encoding) EncodeInt64(dst []byte, src []int64) ([]byte, error) {
-	return append(dst[:0], unsafecast.Int64ToBytes(src)...), nil
+	if cpu.IsBigEndian {
+		srcLen := len(src)
+		byteEnc := make([]byte, (srcLen * 8))
+		idx := 0
+		for k := 0; k < srcLen; k++ {
+			binary.LittleEndian.PutUint64(byteEnc[idx:(8+idx)], uint64((src)[k]))
+			idx += 8
+		}
+		return append(dst[:0], (byteEnc)...), nil
+	} else {
+		return append(dst[:0], unsafecast.Int64ToBytes(src)...), nil
+	}
 }
 
 func (e *Encoding) EncodeInt96(dst []byte, src []deprecated.Int96) ([]byte, error) {
 	return append(dst[:0], deprecated.Int96ToBytes(src)...), nil
 }
 
 func (e *Encoding) EncodeFloat(dst []byte, src []float32) ([]byte, error) {
-	return append(dst[:0], unsafecast.Float32ToBytes(src)...), nil
+	if cpu.IsBigEndian {
+		srcLen := len(src)
+		byteEnc := make([]byte, (srcLen * 4))
+		idx := 0
+		for k := 0; k < srcLen; k++ {
+			binary.LittleEndian.PutUint32(byteEnc[idx:(4+idx)], math.Float32bits((src)[k]))
+			idx += 4
+		}
+		return append(dst[:0], (byteEnc)...), nil
```

Comment:
These seem like the same operations we perform in `column_index.go`, more reasons to move them into functions that can be shared to avoid repeating the code.